### PR TITLE
TIP-613: Sync ES and MySQL

### PR DESCRIPTION
**Pull request by arnolanglade**
Prevent to get a product before the index refreshing. To fix that we use the refresh option: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html with the value 'wait_for'.

### CODE REVIEW

File: src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Akeneo\Bundle\ElasticsearchBundle;
+
+/**
+ * @author    Arnaud Langlade <arnaud.langlade@@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class Refresh
```
Comment (arnolanglade) [inline src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php]: Do we need to this value object to represent the refresh param?

File: src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+{
+    const ENABLE = true;
+    const DISABLE = false;
+    const WAIT_FOR = 'wait_for';
+
+    /** @var bool|string */
+    private $type;
+
+     function __construct($type)
```
Comment (anaelChardan) [inline src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php]: should be private :)

Comment (damien-carcel) [reply src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php]: you mean public?

**Review by anaelChardan — APPROVED** [review]

File: src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php (hunk 2 above)
Comment (damien-carcel) [inline src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php]: phpdoc

Comment (arnolanglade) [reply src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php]: yes it should be private

Comment (anaelChardan) [reply src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php]: No, private because we must use named constructor :)

**Review by damien-carcel — COMMENTED** [review]
Wait to see what the other thinks, but it is OK for me.

File: src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php (hunk 2 above)
Comment (jjanvier) [reply src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php]: private + phpdoc

File: src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * @return Refresh
+     */
+    public static function enableParam()
```
Comment (jjanvier) [inline src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php]: I would remove the "param" suffix to all the methods personnally
I don't understand the meaning

File: src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Akeneo\Bundle\ElasticsearchBundle;
+
+/**
```
Comment (jjanvier) [inline src/Akeneo/Bundle/ElasticsearchBundle/Refresh.php:5]: explain please

and provide link to the ES doc..

**Review by jjanvier — APPROVED** [review]
gtm once fixed

File: src/Akeneo/Bundle/ElasticsearchBundle/spec/RefreshSpec.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+namespace spec\Akeneo\Bundle\ElasticsearchBundle;
+
+use Akeneo\Bundle\ElasticsearchBundle\Refresh;
+use PhpSpec\ObjectBehavior;
+
+class RefreshSpec extends ObjectBehavior
+{
+    function it_creates_a_enable_refresh_param()
+    {
+        $this->beConstructedThrough('enableParam', [Refresh::ENABLE]);
```
Comment (ahocquard) [inline src/Akeneo/Bundle/ElasticsearchBundle/spec/RefreshSpec.php]: the `[Refresh::ENABLE]` param is useless no ?

**Pull request merged**